config/config.go

```diff
@@ -942,8 +946,6 @@ var deprecatedConfig = map[string]struct{}{
 	"max-txn-time-use":                   {},
 	"experimental.allow-auto-random":     {},
 	"enable-redact-log":                  {}, // use variable tidb_redact_log instead
-	"tikv-client.copr-cache.enable":      {},
-	"alter-primary-key":                  {}, // use NONCLUSTERED keyword instead
```

Comment:
I probably miss some context, but why are these removed?

Reply:
> "The "deprecatedConfig" was misnamed, since these options are not really deprecated but actually removed. Or rather it was ambiguous, because some were actually deprecated and still took effect."

They are deprecated, they both still have usage. I initially split this into a map for deprecated and a map for removed, and even wrote tests to make sure they were not in both lists :-) But actually, there is no code that cares about deprecated to do anything different. So I just removed them from the list and renamed it.